control-plane/agents/src/bin/core/controller/scheduling/volume.rs

```diff
@@ -326,8 +330,16 @@ impl GetChildForRemovalContext {
 
 impl DecreaseVolumeReplica {
     async fn builder(request: &GetChildForRemoval, registry: &Registry) -> Result<Self, SvcError> {
+        let mut pool_vg_replica_count_map: Option<HashMap<PoolId, u64>> = None;
+        if let Some(volume_group) = &request.spec.volume_group {
+            if let Ok(volume_group_spec) = registry.specs().volume_group_spec(volume_group.id()) {
```

Comment:
I think here can be an error too:
let volume_group_spec = registry.specs().volume_group_spec(volume_group.id())?;